services/api/src/lib/circle-adapter.ts

```diff
@@ -286,6 +334,63 @@ export default class CircleAdapter {
     return null
   }
 
+  async getTransfersForAddress(
+    query: CircleTransferQuery,
+    destinationAddressId: string
+  ): Promise<ToPaymentBase | null> {
+    const searchParams = {}
+    if (query.walletId)
+      Object.assign(searchParams, { walletId: query.walletId })
+    if (query.sourceWalletId)
+      Object.assign(searchParams, { sourceWalletId: query.sourceWalletId })
+    if (query.destinationWalletId)
+      Object.assign(searchParams, {
+        destinationWalletId: query.destinationWalletId,
+      })
+    if (query.from) Object.assign(searchParams, { from: query.from })
+    if (query.to) Object.assign(searchParams, { to: query.to })
+    if (query.pageBefore)
+      Object.assign(searchParams, { pageBefore: query.pageBefore })
+    if (query.pageAfter)
+      Object.assign(searchParams, { pageAfter: query.pageAfter })
+    if (query.pageSize)
+      Object.assign(searchParams, { pageSize: query.pageSize })
+    const response = await this.http
+      .get('v1/transfers', {
+        searchParams,
+      })
+      .json<CircleResponse<CircleTransfer[]>>()
+
+    if (isCircleSuccessResponse(response)) {
+      const transfer = response.data.find(
+        (transfer) => transfer.destination.address === destinationAddressId
+      )
```

Comment:
Since we're generating a new address for every transaction, there should only be one transfer for that address, though there's no way to search by the address in Circle.